broker/src/test/java/io/camunda/zeebe/broker/exporter/metrics/MetricsExporterTest.java

```diff
@@ -0,0 +1,173 @@
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.exporter.metrics;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.exporter.api.context.Configuration;
+import io.camunda.zeebe.exporter.api.context.Context;
+import io.camunda.zeebe.exporter.api.context.Controller;
+import io.camunda.zeebe.exporter.api.context.ScheduledTask;
+import io.camunda.zeebe.protocol.Protocol;
+import io.camunda.zeebe.protocol.record.ImmutableRecord;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.slf4j.Logger;
+
+class MetricsExporterTest {
+
+  @Test
+  void shouldObserveJobLifetime() {
+    // given
+    final var metrics = new ExecutionLatencyMetrics();
+    final var exporter = new MetricsExporter(metrics);
+    exporter.open(new FakeController());
+    assertThat(metrics.getJobLifeTime().collect())
+        .flatMap(x -> x.samples)
+        .describedAs("Expected no metrics to be recorded at start of test")
+        .isEmpty();
+
+    // when
+    exporter.export(
+        ImmutableRecord.builder()
+            .withRecordType(RecordType.EVENT)
+            .withValueType(ValueType.JOB)
+            .withIntent(JobIntent.CREATED)
+            .withTimestamp(1651505728460L)
+            .withKey(Protocol.encodePartitionId(1, 1))
+            .build());
+    exporter.export(
+        ImmutableRecord.builder()
+            .withRecordType(RecordType.EVENT)
+            .withValueType(ValueType.JOB)
+            .withIntent(JobIntent.COMPLETED)
+            .withTimestamp(1651505729571L)
+            .withKey(Protocol.encodePartitionId(1, 1))
+            .build());
+
+    // then
+    assertThat(metrics.getJobLifeTime().collect())
+        .flatMap(x -> x.samples)
+        .filteredOn(s -> s.name.equals("zeebe_job_life_time_count"))
+        .map(s -> s.value)
+        .describedAs("Expected exactly 1 observed job_life_time sample counted")
+        .containsExactly(1d);
+  }
+
+  private static class FakeController implements Controller {
+
+    @Override
+    public void updateLastExportedRecordPosition(final long position) {
+      // don't care
+    }
+
+    @Override
+    public ScheduledTask scheduleCancellableTask(final Duration delay, final Runnable task) {
+      // don't care
+      return null;
+    }
+  }
+
+  @Nested
+  @DisplayName("MetricsExporter should configure a Filter")
+  class FilterTest {
+
+    static Stream<TypeCombination> acceptedCombinations() {
```

Comment:
:wrench: Clever :slightly_smiling_face: Maybe too clever? :wink: The requirements, I think, are: only events, and then only JOB, JOB_BATCH, PROCESS_INSTANCE? So we could rewrite this as one test checking that only EVENT is accepted (regardless of value type), and one test that only the 3 value types are accepted.

Or is it because we don't want to assume the implementation of `RecordFilter` (i.e. that it treats RecordType and ValueType independently), so we want to test the combinations? Its interface already seem to imply they are treated independently, after all.

Reply:
Ah, well originally I wasn't aware of how the record filter worked. I assumed it would take some combination of record type, valuetype, and intent to filter a single record, but instead it offers this strange inflexible configuration. Apparently, you can't say: only export JOB events and JOB_BATCH commands. I wrote the test playing around with this concept and thought: If we happen to change the exporter API, then this test at least already checks for the correct combinations.

Reply:
I think it makes sense to allow filtering by intent eventually, but hadn't really spent time thinking about more fine-grained control (e.g. allow JOB commands, but not process commands). I'll let the automation team decide here :slightly_smiling_face:

Reply:
I don't necessarily need it now, but I think it makes sense to have this eventually. This test is at least ready for it when it does come 😆